webapp/sources/rudder/rudder-web/src/main/elm/healthcheck/sources/View.elm

```diff
@@ -0,0 +1,118 @@
+module View exposing (..)
+
+import DataTypes exposing (Check, Model, Msg(..), SeverityLevel(..))
+import Html exposing (Html, button, div, i, span, text)
+import Html.Attributes exposing (class)
+import Html.Events exposing (onClick)
+import List exposing (any, map, sortBy, sortWith)
+import String exposing (length)
+
+compareSeverityLevel: Check -> Check -> Order
+compareSeverityLevel a b =
+  case (a.level,b.level) of
+    (Critical, _)     -> GT
+    (Warning, Critical)     -> LT
+    (Warning, CheckPassed)  -> GT
+    (CheckPassed, _)  -> LT
+    (Warning, Warning)  -> LT
+
+
+containsSeverityLevel: SeverityLevel -> List Check -> Bool
+containsSeverityLevel level checks =
+  any (\c -> c.level == level) checks
+
+chooseHigherSecurityLevel: List Check -> SeverityLevel
+chooseHigherSecurityLevel checks =
+  if containsSeverityLevel Critical checks then
+    Critical
+  else if containsSeverityLevel Warning checks then
+    Warning
+  else
+    CheckPassed
+
+
+severityLevelToString: SeverityLevel  -> String
+severityLevelToString level =
+  case level of
+    Warning     -> "warning"
+    CheckPassed -> "ok"
+    Critical    -> "critical"
+
+severityLevelToIcon: SeverityLevel  -> Html Msg
+severityLevelToIcon level =
+  case level of
+    Critical    -> i [class "fa fa-times-circle critical-icon icon-info"][]
+    Warning     -> i [class "fa fa-exclamation-circle warning-icon icon-info"][]
+    CheckPassed -> i [class "fa fa-check-circle ok-icon icon-info"][]
+
+displayCheckListButton: Bool -> Html Msg
+displayCheckListButton isOpen =
+    let
+        classNameBtn = "btn-checks btn btn-sm btn-outline-secondary"
+    in
+    if isOpen then
+      button [class classNameBtn, onClick CheckListDisplay]
+      [
+          text "Hide check's list"
+        , i [class "fa-btn-checks fa fa-chevron-down"][]
+      ]
+    else
+      button [class classNameBtn, onClick CheckListDisplay]
+      [
+          text "Show check's list"
+        , i [class "fa-btn-checks fa fa-chevron-right"][]
+      ]
+
+displayBigMessage: SeverityLevel -> Html Msg
+displayBigMessage level =
+  let
+    levelStr = severityLevelToString level
+  in
+  div[class "global-msg"]
+  [
+    div [class (levelStr++"-info")]
+    [
+        severityLevelToIcon level
+      , case level of
+          Critical ->
+            text "Critical Error"
+          Warning ->
+            text "Should be improved"
+          CheckPassed ->
+            text "All check passed"
+    ]
+  ]
+
+checksDisplay: Model -> List Check -> Html Msg
+checksDisplay model h =
+  let
+    sortedContent = sortBy (\c -> length c.msg) (sortWith compareSeverityLevel h)
```

Comment:
Sort by check name it will be more consistent

Reply:
But here you will sort first by securtylevel then by name/messagelength si you will lose the sort by severity level

Reply:
You should make a comparison on name when your are defining your Order on Check on equality cases in compareSeverityLevel (which should be named, compareCheck, you may need a compareSecurityLevel that compare SecurityLevel only)